hand-written.md

```diff
@@ -1,145 +1,89 @@
 
-There are a number of improvements:
-* Reliable completion, also in partial expressions and syntactically incorrect programs: try `class C { def f(l: List[Int]) = l.<TAB>`
-* CamelCase completion: try `(l: List[Int]).rro<TAB>`, it expands to `(l: List[Int]).reduceRightOption`
-* Show desugarings performed by the compiler by adding `//print`: try `for (x <- 1 to 10) println(x) //print<TAB>`
-* Complete bean getters without typing `get`: try `(d: java.util.Date).day<TAB>`
-* Find members by typing any CamelCased part of the name: try `classOf[String].typ<TAB>` to get `getAnnotationsByType`, `getComponentType` and others
-* Complete non-qualified names, including types: try `def f(s: Str<TAB>`
-* Press tab twice to see the method signature: try `List(1,2,3).part<TAB>`, which completes to `List(1,2,3).partition`; press tab again to display `def partition(p: Int => Boolean): (List[Int], List[Int])`
-
-Thanks to @som-snytt for helping out with this work!
 
 #### New bytecode optimizer
 
-The GenBCode backend includes a new inliner and bytecode optimizer.
+The GenBCode back end includes a new inliner and bytecode optimizer.
 The optimizer is enabled using the `-Yopt:l:classpath` compiler option.
 Check `-Yopt:help` to see the full list of available options for the optimizer.
 
-As of M3, the following optimizations are available:
+As of M4, the following optimizations are available:
 
 * Inlining final methods, including methods defined in objects and final methods defined in traits
 * If a closure is allocated and invoked within the same method, the closure invocation is replaced by an invocations of the corresponding lambda body method
 * Dead code elimination and a small number of cleanup optimizations
+* Box/unbox elimination [#4858](https://github.com/scala/scala/pull/4858)
 
 The work on the new optimizer is still ongoing.  You can track it in the [scala-dev repository issue tracker](https://github.com/scala/scala-dev/labels/t%3Aoptimizer).
 
-#### The `@implicitAmbiguous` annotation
 
-The `@implicitAmbiguous` annotation allows customizing the error message when an implicit search finds multiple ambiguous values. Refer to the [Scaladoc of the annotation class](http://www.scala-lang.org/api/2.12.x/#scala.annotation.implicitAmbiguous) for an example.
+### Breaking changes
 
-Thanks to @puffnfresh for this contribution!
+#### SAM types
+As of [#4971](https://github.com/scala/scala/pull/4971), we treat Single Abstract Method types in the same way as our built-in FunctionN classes. This means overloading resolution has more contenders to choose from, making type inference less effective. Please consult the PR for details.
```

Comment:
this pr is a veeery long and technical discussions, we'll need to have something more accessible for the final release notes